### PR TITLE
 Planning: OpenSpace: Constructing valet parking scenarios and stages 

### PR DESCRIPTION
**Pull request by JasonZhou404**

### CODE REVIEW

File: modules/planning/scenarios/valet_parking/BUILD [hunk 1]
```diff
@@ -6,44 +6,12 @@ cc_library(
     name = "valet_parking",
     srcs = [
         "valet_parking_scenario.cc",
```
Comment (Capri2014) [inline modules/planning/scenarios/valet_parking/BUILD]: break them into small ones

File: modules/planning/scenarios/valet_parking/valet_parking_scenario.cc [hunk 2]
```diff
@@ -17,21 +17,206 @@
...
+    AERROR << "miss scenario specific config";
+    return false;
+  }
+  context_.scenario_config.CopyFrom(config_.valet_parking_config());
+  return true;
 }
 
 bool ValetParkingScenario::IsTransferable(const Scenario& current_scenario,
                                           const Frame& frame) {
+  // TODO(all) Implement avaliable parking spot detection by preception results
```
Comment (Capri2014) [inline modules/planning/scenarios/valet_parking/valet_parking_scenario.cc]: You can add a dummy function "IsParkingSpotAvailable"  and always returns true as of now.

File: modules/planning/scenarios/valet_parking/BUILD (hunk 1 above)
Comment (JasonZhou404) [reply modules/planning/scenarios/valet_parking/BUILD]: Current architect must have them depends on each other

File: modules/planning/scenarios/valet_parking/valet_parking_scenario.cc (hunk 2 above)
Comment (JasonZhou404) [reply modules/planning/scenarios/valet_parking/valet_parking_scenario.cc]: Will do that later, if we don't take available parking spot from routing message

**Pull request merged**